Update site specific information for Bielefeld (05/2021) (#238)
deNBI/cloud-user-docs: wiki/Compute_Center/Bielefeld.md

### Ubuntu apt mirror

We run an apt mirror for Ubuntu LTS releases (18.04 and 20.04) to speed up package download. The mirror is available 
from Bielefeld cloud site from the external (http[s]://apt-cache.bi.denbi.de:9999 or http://129.70.51.2:9999) and c
ebitec (http://172.21.40.2:9999) network.
This mirror is synced every midnight with the official Canonical repositories.

## Object storage

The storage backend used by Bielefeld cloud site is powered by [Ceph](https://www.ceph.com). The Object storage 
endpoint provides API access via SWIFT and S3. The latter should be preferred due to better performance.
You can find a tutorial [here](../Tutorials/ObjectStorage/index.md) on how to use this service.

## Server Groups for optional performance gains

Our OpenStack cluster consists of many compute nodes which hosts all running instances. In some applications, 
it would be reasonable, when multiple instances of your project would be scheduled on as many nodes as possible.

- Distributed systems (like HPC, databases...) can gain a significant performance gain.
- Spread instances over several compute nodes increases the availability when running a high availability setup. 

> This can be achieved with *Server Groups*. Server Groups act as a "container" for instances and it describes 

Reviewer: ```suggestion
This can be achieved with *Server Groups*. Server Groups act as a "container" for instances and describe 
```

| anti affinity     | Force schedule all instances as spread as possible on all compute nodes.     |
| soft anti affinity    | Try to schedule all instances as spread as possible on all compute nodes. Allow to violate this policy when there are not enough compute nodes with such capacity.     |

It is recommended to use the "soft" variant. Otherwise, instances can fail to start when they would violate 
the more strict policy options.

Afterwards the creation of a new Server Group, you can add instances to it when you are creating them. 
It's not possible to add already running instances to a Server Group since they are already scheduled.
On the *Server Groups* tab, add the group by clicking on the small up-arrow:
![sg_screen3](img/bielefeld/sg_screen3.png)

Afterwards, the scheduling of this instance will respect your selected Server Group policy.

## Application Credentials (use OpenStack API)

> In order to access the OpenStack Cloud via commandline tools, you need to source a so called rc file as 
> described [here](https://cloud.denbi.de/wiki/Tutorials/ObjectStorage/#retrieving-access-credentials).
> However, the standard procedure does not work on all Cloud locations. Executing `source` on the 
> downloaded rc file prompts for a password. This password **is not the same** you have used when 
> authenticating to ELIXIR in order to access the OpenStack Dashboard.

Reviewer: ```suggestion
In order to access the OpenStack Cloud via command-line tools, you need to source a so called rc file as 
described [here](https://cloud.denbi.de/wiki/Tutorials/ObjectStorage/#retrieving-access-credentials).
However, the standard procedure does not work on all Cloud locations. Executing `source` on the 
downloaded rc file prompts for a password. This password **is not the same** you have used when 
authenticating to ELIXIR in order to access the OpenStack Dashboard.
```

Internally, OpenStack does not set a local password for your ELIXIR-ID, since it does not need to 
hence OpenStack confirms your authorization separately via ELIXIR AAI.
However, the commandline-tools can only function with a set local password. Prior to the new 
OpenStack release, users had to contact the cloud site administrators in order for them to set an 
explicit local password and send it back to the user via encrypted mail or de.NBI vault service. 

> Luckily, there is a new feature since OpenStack Rocky where the user is able to set its own *local* 

Reviewer: ```suggestion
Luckily, there is a new feature since OpenStack Rocky where users are able to set their own *local* 
```

Now you can use the openstack commandline tools.

### Note

Application credentials are currently not supported by all applications or development kits accessing the 

> Openstack API. If this is case users had to contact the cloud site administrators in order for them to set 

Reviewer: ```suggestion
Openstack API. In this case users have to contact the cloud site administrators in order for them to set 
```

Detaching the floating IP will also delete the A record.

## (Information) Security

Our [information security policy (german language)](assets/bielefeld/informationssicherheitsleitlinie.md)  
ensures that we follow defined protocols and procedures.

### Clock synchronisation

Our cloud infrastructure and user virtual machines are clock synchronised using the Network Time Protocol (NTP).

### Information security incidents

> If users get aware of a security incident or notice an abnormal behaviour of his instance (unexpected high CPU 
> load or high network traffic without running actively anything), he should immediately get in contact with 
> the cloud administrators (os-service@cebitec.uni-bielefeld.de). The administrators will help the user to separate 
> the concerned instance(s) from the public network and will start a basic forensic analyses (with the users' 
> agreement). Depending on the analysis result a security incident report is written and the security incident 
> officer of Bielefeld university is informed. If any personal data is affected, the data security officer 
> is also consulted.

Reviewer: ```suggestion
If users become aware of a security incident or notice an abnormal behaviour of their instances (e.g. unexpected high CPU 
load or high network traffic without actively running anything), they should immediately contact 
the cloud administrators (os-service@cebitec.uni-bielefeld.de). The administrators will assist in isolating
the affected instances from the public network and will (with the owner's consent) start a basic forensic analysis.
Depending on the analysis results a security incident report is written and the security incident 
officer of Bielefeld university is informed. If any personal data is affected, the data security officer 
is also consulted.
```

> ### Cryptographic 
> 
> We use cryptographic methods to protect the users' data located on our infrastructure :
> 
> - Openstack UI and Openstack API is secured by TLS
> - VMs local discs are located on encrypted devices (LUKS).
> - Cloud storage  is also LUKS encrypted (work in progress)

Reviewer: ```suggestion
### Cryptography

We use cryptographic methods to protect user data stored on our infrastructure:

- Client connections to our OpenStack Dashboard and Openstack API 
  are encrypted using TLS
- Local disks of instances are located on encrypted devices (LUKS).
- Cloud storage  is LUKS encrypted (work in progress)
```
